lib/ZnapZend/ZFS.pm

```diff
@@ -122,6 +123,30 @@ sub snapshotExists {
     return grep { $snapshot eq $_ } @snapshots;
 }
 
+sub createDataSet {
+    my $self = shift;
+    my $dataSet = shift;
+    my $remote;
+
+    #just in case if someone aks to check '';
+    return 0 if !$dataSet;
+
+    ($remote, $dataSet) = $splitHostDataSet->($dataSet);
+    my @ssh = $self->$buildRemote($remote,
+        [@{$self->priv}, qw(zfs create), $dataSet]);
+
+    print STDERR '# ' . join(' ', @ssh) . "\n" if $self->debug;
+
+    #return if 'noaction' or snapshot creation successful
+    return 1 if $self->noaction || !system(@ssh);
+
+    #check if snapshot already exists and therefore creation failed
+    return 0 if $self->dataSetExists($dataSet);
+
```

Comment:
the code below seems to imply that we only try to create the dataset if it does not exist ... sooo